content_scripts/link_hints.coffee

```diff
@@ -673,5 +674,25 @@ class TypingProtector extends Mode
 
     @onExit callback
 
+# TODO(mrmr1993): Use event.deepPath to traverse into shadow DOMs, as part of a solution to #1861. This
```

Comment:
> We have appended target node's ancestor to `registrationElement` if `document.contains` returns `false`

Only if `document.parentNode` is `null`, which is never the case for elements in a shadow DOM. This is intentional, so that we don't cause re-renders or trigger `MutationObserver`s by detatching and re-attaching shadow dom elements.